src/api_server/swagger/firecracker.yaml

```diff
@@ -668,6 +751,46 @@ definitions:
         $ref: "#/definitions/TokenBucket"
         description: Token bucket with operations as tokens
 
+  CreateSnapshotParams:
+    type: object
+    required:
+      - snapshot_path
+      - mem_file_path
+    properties:
+      snapshot_path:
+        type: string
+        description: Path to the file that will contain the microVM state.
+      mem_file_path:
+        type: string
+        description: Path to the file that will contain the guest memory.
+      snapshot_type: string
+        enum:
+          - Full
+          - Diff
+        description:
+          Type of snapshot to create. It is optional and by default, a full
+          snapshot is created.
```

Comment:
fyi: one of the topics of discussions with Lambda customers when we reviewed the API proposal was that all these knobs become hard to manage. That is why `/snapshot/create` and `/snapshot/create-diff` were ultimately different api endpoints (to simplify this config body).

In its current format you also need to document the behavior of `snapshot_type == Diff` in rapport with `enable_diff_snapshots` from a previous snapshot load or `enable_diff_snapshots` from vm-config in case this is a _fresh_ microvm.

Reply:
This also brings up some extra checks required when validating the request.Does it make sense to make the simplifying assumption that fresh microvms cannot create incremental snapshots ?

Reply:
> Does it make sense to take the simplifying assumption that fresh microvms cannot create incremental snapshots ?

No...
- a `full` snapshot on a fresh microvm means a full memory file of size `==` to guest memory
- a `diff` snapshot on a fresh microvm means a sparse memory file containing only written pages.